source/include/FreeRTOS_IPv6_Private.h

```diff
@@ -28,20 +28,16 @@
 #ifndef FREERTOS_IPV6_PRIVATE_H
 #define FREERTOS_IPV6_PRIVATE_H
 
-/* Application level configuration options. */
-#include "FreeRTOSIPConfig.h"
-#include "FreeRTOSIPConfigDefaults.h"
+/* Global Includes & Definitions */
 #include "FreeRTOS_IP_Common.h"
+
+/* Core FreeRTOS+TCP Includes. */
 #include "FreeRTOS_Sockets.h"
 #include "FreeRTOS_Stream_Buffer.h"
-#if ( ipconfigUSE_TCP == 1 )
-    #include "FreeRTOS_TCP_WIN.h"
-    #include "FreeRTOS_TCP_IP.h"
-#endif
-
-#include "semphr.h"
 
-#include "event_groups.h"
+/* Optional FreeRTOS+TCP Includes. */
+#include "FreeRTOS_TCP_WIN.h"
+#include "FreeRTOS_TCP_IP.h"
```

Comment:
```suggestion
#if ( ipconfigUSE_TCP == 1 )
    #include "FreeRTOS_TCP_WIN.h"
    #include "FreeRTOS_TCP_IP.h"
#endif
```
We should align the behavior on this header.

Reply:
Okay I can do that if you prefer, I had mentioned in my original post that the reason I got rid of them was I figured it was cleaner to just guard the source implementation with the config macros rather than try to catch every include everywhere. You aren't wasting flash if the function declarations are never defined.

Reply:
Could you show me the original post? I'd like to align the information first.